### ISSUE TITLE
`:=` and `:=!` too losely bound

### ISSUE THREAD

**Issue by 3noch** [PkmX/rawr#4]
I'm not sure it's even possible to fix this but you'll notice that when defining a record with function types, you must use parenthesis like this:

```haskell
type MyRec = R ( "runThing" := (a -> IO b) )
```

Without parens, the parser thinks you are defining `("runThing" := a) -> IO b` as an unlabeled record.

**Comment by PkmX** [comment]
I don't think this is fixable, since `(->)` is hard-wired as `infixr 0` in GHC. As you mentioned, adding parentheses around would be the easiest workaround of this issue.
